controllers/utils/multicloud_util.go

```diff
@@ -0,0 +1,341 @@
+		string(installPlanClusterServiceVersionNames))
+
+	actionSpec := actionv1beta1.ActionSpec{
+		ActionType: actionv1beta1.UpdateActionType,
+		KubeWork: &actionv1beta1.KubeWorkSpec{
+			Resource:       "installplan",
+			Namespace:      installPlan.ObjectMeta.Namespace,
+			ObjectTemplate: runtime.RawExtension{Raw: []byte(templateContent)},
+		},
+	}
+
+	return actionSpec, nil
+}
+
+// CheckManagedClusterActionDone returns error for all the cases in which an action didn't manage to complete
+// successfully.
+func CheckManagedClusterActionDone(
+	ctx context.Context, c client.Client, mcaName string, mcaNamespace string, Log logr.Logger) error {
+	// Check the managedClusterAction has completed successfully.
+	mca := &actionv1beta1.ManagedClusterAction{}
+	err := c.Get(ctx, types.NamespacedName{Name: mcaName, Namespace: mcaNamespace}, mca)
+	if err != nil {
+		return err
+	}
+	// Check that the ManagedClusterAction has completed.
+	conditionMCA := meta.FindStatusCondition(mca.Status.Conditions, actionv1beta1.ConditionActionCompleted)
+	if conditionMCA == nil {
+		err = fmt.Errorf("ManagedClusterAction hasn't completed yet")
+		return err
+
+	}
+	// Check that the ManagedClusterAction has completed successfully.
+	if conditionMCA.Reason != ManagedClusterActionReasonDone {
+		if conditionMCA.Reason == actionv1beta1.ReasonUpdateResourceFailed {
+			err = fmt.Errorf("ManagedClusterAction failed: %s", conditionMCA.Message)
+			return err
+		}
+		err = fmt.Errorf("ManagedClusterAction has unexpected condition reason: %s", conditionMCA.Reason)
+		return err
+	}
+
+	return nil
+}
+
+// DeleteMultiClusterObjects cleans up views associated to an UOCR.
+func DeleteMultiClusterObjects(
+	ctx context.Context, c client.Client, clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, Log logr.Logger) error {
+	// Only need to delete ManagedClusterView as ManagedClusterActions get deleted automatically.
+	for _, batch := range clusterGroupUpgrade.Status.RemediationPlan {
+		for _, clusterName := range batch {
+			var mcvLabels = map[string]string{
+				"openshift-cluster-group-upgrades/clusterGroupUpgrade": clusterGroupUpgrade.Name}
+			opts := []client.ListOption{
+				client.InNamespace(clusterName),
+				client.MatchingLabels(mcvLabels),
+			}
+
+			mcvList := &viewv1beta1.ManagedClusterViewList{}
+			if err := c.List(ctx, mcvList, opts...); err != nil {
+				return err
+			}
+
+			for _, mcv := range mcvList.Items {
+				Log.Info("[DeleteMultiClusterObjects] Delete ManagedClusterView", "name", mcv.Name)
+				if err := c.Delete(ctx, &mcv); err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// GetMultiCloudObjectName computes the name of a view or action
+func GetMultiCloudObjectName(managedClusterType string, clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, kind string, objectName string) string {
+	return strings.ToLower(managedClusterType + "-" + clusterGroupUpgrade.Name + "-" + kind + "-" + objectName)
```

Comment:
If there are two CGU with the same name in different namespaces they could generate the same named MCA or MCV. If you add the CGU namespace then that small case is avoided.